interface/FrankWriter.h

```diff
@@ -86,7 +91,7 @@ template <class EventClass> class FrankWriter : public BaseOperator<EventClass>
       eventInfo = dynamic_cast<mut::EventInfo *>(&ev.eventInfo_);
```

Comment:
you have to add a dynamic cast to redirect after each event, i.e., best_match_distances_ptr_ = dynamic_cast<std::vector<std::vector <double>> *>(&ev.best_match_distances_)